series/series.go

```diff
@@ -740,3 +746,186 @@ func (s Series) Quantile(p float64) float64 {
+		}
+	}
+	bin := func(n float64) float64 {
+		return math.Floor(n - float64(int(n)%int(binwidth)))
+	}
+	for _, v := range s.Float() {
+		h[bin(v)]++
+	}
+	return h
+}
+
+// Median calculates the middle or median value, as opposed to
+// mean it is not susceptible to being affected by outliers.
+func (s Series) Median() float64 {
+	if s.elements.Len() == 0 ||
+		s.Type() == String ||
+		s.Type() == Bool {
+		return math.NaN()
+	}
+	ix := s.Order(false)
+	newElem := make([]Element, len(ix))
+
+	for newpos, oldpos := range ix {
+		newElem[newpos] = s.elements.Elem(oldpos)
+	}
+
+	// When length is odd, we just take length(list)/2
+	// value as the median.
+	if len(newElem)%2 != 0 {
+		return newElem[len(newElem)/2].Float()
+	}
+	// When length is even, we take middle two elements of
+	// list and the median is an average of the two of them.
+	return (newElem[(len(newElem)/2)-1].Float() +
+		newElem[len(newElem)/2].Float()) * 0.5
+}
+
+func sum(s []float64) float64 {
+	if len(s) > 1 {
+		return s[0] + sum(s[1:])
+	}
+	return s[0]
+}
+
+// Sum calculates sum of given series
+func (s Series) Sum() float64 {
+	if s.elements.Len() == 0 || s.Type() == String {
+		return math.NaN()
+	}
+	return sum(s.Float())
+}
+
+func applyFloat(s1, s2 Series, op int) interface{} {
+	delta := make([]float64, s1.Len())
+	for i := 0; i < s1.Len(); i++ {
+		left := s1.elements.Elem(i).Val().(float64)
+		right := s2.elements.Elem(i).Val().(float64)
+		if math.IsNaN(left) || math.IsNaN(right) {
+			delta[i] = math.NaN()
+			continue
+		}
+		if op == Add {
+			delta[i] = left + right
+		} else if op == Subtract {
+			delta[i] = left - right
+		} // Maybe other operations should be supported
+	}
+	return delta
+}
+
+func applyInt(s1, s2 Series, op int) interface{} {
+	delta := make([]int, s1.Len())
+	for i := 0; i < s1.Len(); i++ {
+		left := s1.elements.Elem(i).Val().(int)
+		right := s2.elements.Elem(i).Val().(int)
+		if op == Add {
+			delta[i] = left + right
+		} else if op == Subtract {
+			delta[i] = left - right
+		} // Maybe other operations should be supported
+	}
+	return delta
+}
+
+func elemWiseAdd(s1, s2 Series, t Type,
+	f func(a Series, b Series, op int) interface{}) Series {
+	return New(f(s1, s2, 0), t, "")
+}
+
+func elemWiseSub(s1, s2 Series, t Type,
+	f func(a Series, b Series, op int) interface{}) Series {
+	return New(f(s1, s2, 1), t, "")
+}
+
+// ElementWiseSubtract will piece-wise subtract elements
+// in s2 Series from elements in s.
+func (s Series) ElementWiseSubtract(s2 Series) Series {
+	if s.elements.Len() == 0 {
+		return s
+	}
+	if err := s.Err; err != nil {
+		return s
+	}
+	if s.Type() != s2.Type() {
+		return Series{}
+	}
+	if s.Len() != s2.Len() {
+		return s
+	}
+	switch s.t {
+	case Float:
+		return elemWiseSub(s, s2, Float, applyFloat)
+	case Int:
+		return elemWiseSub(s, s2, Int, applyInt)
+
+	default:
+		panic(fmt.Sprintf("subtraction operation not supported on %v type", s.t))
```

Comment:
Try to avoid using panic() unless is necessary. Default to return a Series containing an error consistently with the rest of the library.